### PR TITLE
ROCm jenkins run distributed tests separately as new test3

### PR DESCRIPTION
**Pull request by jeffdaily**
CC @ezyang @xw285cornell @sunway513 

Corresponds to https://github.com/pytorch/ossci-job-dsl/pull/65

### CODE REVIEW

**Comment by ezyang** [comment]
I think this is affecting non-ROCm builds. Not all of our configs do three test configs, which means it would be invalid to exclude things from test2, since test3 will never get run.

**Review by ezyang — CHANGES_REQUESTED** [review]
I don't think this works

**Comment by xw285cornell** [comment]
qq: does circleci also uses the same test script? I thought it's ok because it's under the jenkins folder. If so, we can add a rocm guard for test2/3?

**Comment by jeffdaily** [comment]
Does circleci use the same test script?

**Comment by ezyang** [comment]
CircleCI uses this script (it is confusingly named)

File: .jenkins/pytorch/test.sh [hunk 1]
```diff
@@ -308,12 +335,15 @@ elif [[ "${BUILD_ENVIRONMENT}" == *-test2 || "${JOB_BASE_NAME}" == *-test2 ]]; t
   test_libtorch
   test_custom_script_ops
   test_torch_function_benchmark
+elif [[ "${BUILD_ENVIRONMENT}" == *-test3 || "${JOB_BASE_NAME}" == *-test3 ]]; then
```
Comment (ezyang) [inline .jenkins/pytorch/test.sh:338]: `JOB_BASE_NAME`? If this is to deal with Jenkins differences, don't you have to apply this everywhere else?

**Comment by ezyang** [comment]
Please give evidence that: (1) non-ROCm jobs are still running distributed tests and (2) ROCm jobs are not running distributed tests

**Comment by jeffdaily** [comment]
ROCm kernel was patched on CI hosts.  This PR might not be needed any longer since the sporadic mem access faults should be resolved with the patched kernel.  This PR was only creating the test3 to separate out the mGPU tests that seemed to exercise the fault more frequently than sGPU tests.

**Comment by jeffdaily** [comment]
If the need arises again to have a test3, we will open a new PR.  As of today, ROCm CI has stabilized.

**Pull request closed without merging**